### PR TITLE
Make byte array data accessor use RealmAwareZkClient

### LINKED ISSUE #1479: Helix Rest's ZooKeeper endpoints are not ZooScalability compliant
### Describe the bug
The zookeeper endpoints (ZooKeeperAccessor) haven't been migrated to ZooScalabilitiy.

### To Reproduce
Fully reproducible.

### Expected behavior
ZooKeeperAccessor should use a RealmAwareZkClient with a byte array serializer.

### PR DESCRIPTION
**Pull request by narendly**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fixes #1479

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This change was left out of the ZooScalability migration of helix-rest, making ZooKeeperAccessor endpoints fail in a multi-zk setting. This change fixes this.

### Tests

- [x] The following tests are written for this issue:

Tests already exist for this part of the code.

- [x] The following is the result of the "mvn test" command on the appropriate module:

helix-rest module:
```

[INFO] Tests run: 168, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 33.257 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 168, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  38.829 s
[INFO] Finished at: 2020-10-21T17:08:58-07:00
[INFO] ------------------------------------------------------------------------
```

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java [hunk 1]
```diff
@@ -157,6 +158,50 @@ public RealmAwareZkClient getRealmAwareZkClient() {
     return _zkClient;
   }
 
+  /**
+   * Returns a RealmAWareZkClient with ByteArraySerializer with double-checked locking.
+   * NOTE: this is different from getRealmAwareZkClient in that it does not reset listeners for
+   * _zkClientForListener because this RealmAwareZkClient is independent from routing data changes.
+   * @return
+   */
+  public RealmAwareZkClient getByteArrayRealmAwareZkClient() {
```
Comment (dasahcc) [inline helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java:167]: Just have one suggestion. There are many duplicated pieces with previous function. Can we abstract it as a private function and called in both public method?

Comment (jiajunwang) [reply helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java:167]: +1, the only difference is the serializer, right?

Comment (narendly) [reply helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java:135]: I agree with the suggestions made here. This PR has been updated with the refactor.

**Review by dasahcc — APPROVED** [review]

**Comment by narendly** [comment]
This PR is ready to be merged

"This change was left out of the ZooScalability migration of helix-rest, making ZooKeeperAccessor endpoints fail in a multi-zk setting. This change fixes this."

**Pull request merged**